lib/LANraragi/Model/Upload.pm

```diff
@@ -113,8 +113,12 @@ sub handle_incoming_file {
     move( $tempfile, $output_file . ".upload" );
```

Comment:
If the move were to fail due to permission errors on the content folder, I believe you should check the output of this first move operation instead, as it's the one that actually places the file in the correct spot.  

We could probably return early here and not bother with the second move if this one fails too.